static/scripts/rewards/web3/verify-current-network.ts

```diff
@@ -10,7 +10,8 @@ export async function verifyCurrentNetwork(desiredNetworkId: number) {
     return;
   }
 
-  const web3provider = new ethers.providers.Web3Provider(window.ethereum);
+  let web3provider = new ethers.providers.Web3Provider(window.ethereum);
+  web3provider = new ethers.providers.Web3Provider(web3provider.provider);
```

Comment:
The flow without a page reload would be like this:

1. page renders and we call this function.
2. Assume no network change is required, nothing happens.
-
2. Assume network change is required, the change happens after this in another function.
3. We do not update our global signer and this function would only be called again if they clicked claim again, which isn't possible with the current setup as it'll have a loader. Plus these objects are in-memory and not used for txs or anything.
4. The user claim event proceeds and fails against the old provider.

I think a better approach might be for this is to be placed inside `connect-wallet()` which is the place where the signer comes from which is what we are interested in. `connect-wallet` is called prior to any claim which makes taking the newly updated web3Provider/signer easier.

Reply:
@Keyrxng Did you mean not only `verify-current-network.ts` but also apply it to `connect-wallet.ts`?